packages/headless/src/controllers/facets/range-facet/date-facet/headless-date-filter.ts

```diff
@@ -0,0 +1,203 @@
+   *
+   * @defaultValue `true`
+   */
+  filterFacetCount?: boolean;
+
+  /**
+   * The maximum number of results to scan in the index to ensure that the facet lists all potential facet values.
+   *
+   * Note: A high injectionDepth may negatively impact the facet request performance.
+   *
+   * Minimum: `0`
+   *
+   * @defaultValue `1000`
+   */
+  injectionDepth?: number;
+}
+
+export interface DateFilterRange {
+  /**
+   * The starting value for the date range.
+   */
+  start: string;
+
+  /**
+   * The ending value for the date range.
+   */
+  end: string;
+}
+
+export interface DateFilterProps {
+  /**
+   * The options for the `DateFilter` controller.
+   */
+  options: DateFilterOptions;
+  /**
+   * The initial range that should be applied to the `DateFilter` controller.
+   */
+  initialState?: DateFilterRange;
+}
+
+export interface DateFilterState {
+  /**
+   * The facet ID.
+   * */
+  facetId: string;
+
+  /**
+   * The current selected range.
+   */
+  range?: DateFacetValue;
+
+  /**
+   * Returns `true` if a search is in progress, and `false` if not.
+   */
+  isLoading: boolean;
+}
+
+/**
+ * The `DateFilter` controller makes it possible to create a date filter.
+ */
+export interface DateFilter extends Controller {
+  /**
+   * Deselects current filter.
+   */
+  deselect(): void;
+
+  /**
+   * Updates the selected range.
+   *
+   * @param range - The date range.
+   */
+  setRange(range: DateFilterRange): void;
+
+  /**
+   * The state of the `DateFacet` controller.
+   */
+  state: DateFilterState;
+}
+
+export function buildDateFilter(
+  engine: SearchEngine,
+  props: DateFilterProps
+): DateFilter {
+  if (!loadDateFilterReducer(engine)) {
+    throw loadReducerError;
+  }
+
+  const controller = buildController(engine);
+  const {dispatch} = engine;
+  const getState = () => engine.state;
+  const facetId = determineFacetId(engine, props.options);
+  const options: RegisterDateFacetActionCreatorPayload = {
+    ...props.options,
+    currentValues: props.initialState
+      ? [{...props.initialState, endInclusive: true, state: 'selected'}]
+      : [],
+    generateAutomaticRanges: false,
+    facetId,
+  };
+
+  validateDateFacetOptions(engine, options);
+  dispatch(registerDateFacet(options));
+
+  return {
+    ...controller,
+    deselect: () => {
+      dispatch(deselectAllFacetValues(facetId));
+      dispatch(updateFacetOptions({freezeFacetOrder: true}));
+      dispatch(executeSearch(logFacetClearAll(facetId)));
+    },
+    setRange: (range) => {
+      const facetValue: DateFacetValue = {
+        ...range,
+        state: 'selected',
+        numberOfResults: 0,
+        endInclusive: true,
+      };
+
+      dispatch(
+        updateDateFacetValues({
+          facetId,
+          values: [facetValue],
+        })
+      );
+      dispatch(updateFacetOptions({freezeFacetOrder: true}));
+      dispatch(
+        executeSearch(
+          logFacetSelect({
```

Comment:
I don't see the facet slider more appropriate, this method is selecting a range, like any other range values, it's just that this time it's a direct input from the user instead of a manual range or an automatic index one.

This controller is also not technically ready to be a slider, we might need a min/max value to know what extremes to slide into.